### ISSUE TITLE
Stream output is not rendered correctly in `jupyter-collaboration` 3

### ISSUE THREAD

**Issue by trungleduc** [jupyterlab/jupyterlab#16934]
## Description

By using Jupyter Collaboration 3  with JupyterLab 4.3.0, the stream output of a saved notebook will be rendered in one column.

https://github.com/user-attachments/assets/5fe9f989-af1a-4493-9267-4f4ba237d4de

## Reproduce

- Create an environment with JupyterLab 4.3.0 and Jupyter Collaboration 3.0.0
- Create a new notebook with the following content:
```python
text = """Lorem ipsum dolor sit amet, consectetur adipiscing elit. Nullam urna
libero, dictum a egestas non, placerat vel neque. In imperdiet iaculis fermentum. 
Vestibulum ante ipsum primis in faucibus orci luctus et ultrices posuere cubilia 
Curae; Cras augue tortor, tristique vitae varius nec, dictum eu lectus. Pellentesque 
id eleifend eros. In non odio in lorem iaculis sollicitudin. In faucibus ante ut 
arcu fringilla interdum. Maecenas elit nulla, imperdiet nec blandit et, consequat 
ut elit."""
print(text)
```
- Save the notebook then restart JupyterLab
- Remove all hidden cache: 
```bash
 rm -fr .ipynb_checkpoints. .jupyter_ystore.db
```
- Start JupyterLab and open the above notebook

## Expected behavior

 The culprit is in this function: https://github.com/jupyterlab/jupyterlab/blob/main/packages/outputarea/src/model.ts#L514
```js
  export function normalize(value: nbformat.IOutput): void {
    if (nbformat.isStream(value)) {
      if (Array.isArray(value.text)) {
        value.text = (value.text as string[]).join('\n');
      }
    }
  }
```
Since in RTC mode, the output text is a list of characters, and this `normalize` function appends `\n` to all of them. Removing the `'\n'` argument of the `join` function fixes the issue.

## Context

```bash
jupyter-collaboration     3.0.0              pyhd8ed1ab_0    conda-forge
jupyter-collaboration-ui  1.0.0              pyhd8ed1ab_0    conda-forge
jupyter-docprovider       1.0.0              pyhd8ed1ab_0    conda-forge
jupyter-lsp               2.2.5              pyhd8ed1ab_0    conda-forge
jupyter_client            8.6.3              pyhd8ed1ab_0    conda-forge
jupyter_core              5.7.2              pyh31011fe_1    conda-forge
jupyter_events            0.10.0             pyhd8ed1ab_0    conda-forge
jupyter_server            2.14.2             pyhd8ed1ab_0    conda-forge
jupyter_server_fileid     0.9.2              pyhd8ed1ab_0    conda-forge
jupyter_server_terminals  0.5.3              pyhd8ed1ab_0    conda-forge
jupyter_server_ydoc       1.0.0              pyhd8ed1ab_0    conda-forge
jupyter_ydoc              3.0.0              pyhd8ed1ab_0    conda-forge
jupyterlab                4.3.0              pyhd8ed1ab_0    conda-forge
jupyterlab_pygments       0.3.0              pyhd8ed1ab_1    conda-forge
jupyterlab_server         2.27.3             pyhd8ed1ab_0    conda-forge
```

<details><summary>Troubleshoot Output</summary>
<pre>
Paste the output from running `jupyter troubleshoot` from the command line here.
You may want to sanitize the paths in the output.
</pre>
</details>

<details><summary>Command Line Output</summary>
<pre>
Paste the output from your command line running `jupyter lab` here, use `--debug` if possible.
</pre>
</details>

<details><summary>Browser Output</summary>

<pre>
Paste the output from your browser Javascript console here, if applicable.

</pre>
</details>

**Comment by davidbrochart** [comment]
This has been fixed in jupyter-ydoc [v3.0.2](https://github.com/jupyter-server/jupyter_ydoc/releases/tag/v3.0.2).

**Comment by trungleduc** [comment]
Thanks @davidbrochart
